network/src/inbound/cache.rs

```diff
@@ -19,32 +19,35 @@ use crate::Payload;
 use snarkvm_dpc::block::BlockHeader;
 
 use circular_queue::CircularQueue;
+use tokio::sync::Mutex;
 use twox_hash::xxh3::hash64;
 
 pub struct Cache {
-    queue: CircularQueue<u64>,
+    queue: Mutex<CircularQueue<u64>>,
```

Comment:
what's the point of drilling the mutex in? it makes it so we can't have a single-threaded cache, but we practically always lock anyways. I'm also depending on using this construct in a single-thread context in block change PR.

Reply:
that one didn't result in too much difference (just the hash calculation is outside of the scope of the lock now, and the hash is super fast), so I'm ok with reverting this.